packages/react-charting/src/components/DonutChart/DonutChartRTL.test.tsx

```diff
@@ -0,0 +1,198 @@
+import { render, screen, queryAllByAttribute, fireEvent, act } from '@testing-library/react';
+import { chartPoints } from './DonutChart.test';
+import { DonutChart } from './index';
+import * as React from 'react';
+import { DarkTheme } from '@fluentui/theme-samples';
+import { ThemeProvider } from '@fluentui/react';
+import * as utils from '../../utilities/utilities';
+
+test('Should hide callout on mouse leave', () => {
+  // Arrange
+  const { container } = render(<DonutChart data={chartPoints} innerRadius={55} calloutProps={{ doNotLayer: true }} />);
+
+  // Act
+  const getById = queryAllByAttribute.bind(null, 'id');
+  fireEvent.mouseOver(getById(container, /Pie/i)[0]);
+  expect(getById(container, /callout/i)[0]).toBeDefined();
+  fireEvent.mouseLeave(getById(container, /Pie/i)[0]);
+
+  // Assert
+  expect(getById(container, /callout/i)[0]).toHaveStyle('opacity: 0');
+  expect(container).toMatchSnapshot();
+});
+
+test('Should show callout on focus', () => {
+  // Arrange
+  const { container } = render(<DonutChart data={chartPoints} innerRadius={55} calloutProps={{ doNotLayer: true }} />);
+
+  // Act
+  const getById = queryAllByAttribute.bind(null, 'id');
+  fireEvent.focus(getById(container, /Pie/i)[0]);
+
+  // Assert
+  expect(getById(container, /focusRing/i)).toBeDefined();
+});
+
+test('Should remove focus on blur', () => {
+  // Arrange
+  const { container } = render(<DonutChart data={chartPoints} innerRadius={55} calloutProps={{ doNotLayer: true }} />);
+
+  // Act
+  const getById = queryAllByAttribute.bind(null, 'id');
+  fireEvent.blur(getById(container, /Pie/i)[0]);
+
+  // Assert
+  const value = getById(container, /Pie/i)[0].getAttribute('id');
+  expect(value).not.toContain('focusRing');
+});
+
+test('Should highlight the corresponding Pie on mouse over on legends', () => {
+  // Arrange
+  const { container } = render(<DonutChart data={chartPoints} innerRadius={55} hideLegend={false} />);
+
+  // Act
+  const legend = screen.queryByText('first');
```

Comment:
do we need to add a check to ensure legend is defined

Reply:
added   expect(legend).toBeDefined(); after all such queried legends.